### DISCUSSION TITLE
How to properly create a release from a commit id while keeping messages?

### DISCUSSION THREAD

**Discussion by keeper32** [gogs/gogs#7823] [Q&A]
Hello, greetings from Brazil. Can you help me on a small/trivial/newbie question?

I'm used to create releases using gogs web interface, where it allows me to pick a branch and input a name, like `develop` and `3.5.2124`

Also, I can put some description in markdown:
```
- Implements feature 12345.
- Fixes issue 54321.
```

But my develop branch moved foward and now I need to achieve the above result _based on a commit ID instead of the top_, and that's were the issues arise:
- I cant find a way to create a tag/release pointing out a commit ID in the web interface.
- If I push a proper tag containing the code and messages, it appears as a release (with a strangely smaller font), but the tag messages are lost and I'm unable to edit it to add it back.

In another discussion someone said I can create a release from a tag, but I'm at a loss here. 
Any help?

So many edits because of my terrible english, sorry.

**Comment by ahmed-ismail-nt** [comment]
> In another discussion someone said I can create a release from a tag, but I'm at a loss here. Any help?

Assume you are creating release v4.1.0 via tags. You would follow these steps.

- Create a tag locally:
`$ git tag -a 4.1.0 <some-commit-hash-here-if-needed>`
- Push your tag to gogs
`$ git push origin 4.1.0`
- Create a release using the same tag you just pushed. There is a hint below the version field confirming that this is the way to do it as well.
<img width="453" alt="image" src="https://github.com/user-attachments/assets/2c1a4528-addb-433c-8f6b-920ca6203d07" />

- Fill in the release title, comment, etc. as you wish and publish the release.
